x-pack/plugins/observability/public/components/custom_threshold/components/preview_chart/painless_tinymath_parser.ts

```diff
@@ -0,0 +1,205 @@
+      if (part.includes('&&')) {
+        return part.replace('&&', '*').trim();
+      }
+
+      // Handle the NOT and NOT_EQUAL cases and replace them with == and !=
+      let isNOT = false;
+      if (part.includes(this.NOT)) isNOT = true;
+      if (part.includes(this.NOT_EQUAL)) {
+        part = part.replace('!=', '==');
+      }
+      // Strip the part from the parentheses and the NOT characters
+      const stripedPart = part
+        .replace(this.notRegex, '')
+        .replace(this.openParenthesesRegex, '')
+        .replace(this.closeParenthesesRegex, '')
+        .replace(this.NOT, '')
+        .trim();
+
+      // Build the ifelse function and switch the true/false position if it is a NOT case
+      // Every condition will be evaluated to 0 or 1 and then we will use the ifelse function to return the correct value
+      // example: ifelse(A > 0, 1, 0) example for NOT: ifelse(A > 0, 0, 1)
+      let result = `ifelse(${stripedPart},${isNOT ? '0,1' : '1,0'})`;
+
+      const openParentheses = part.match(this.openParenthesesRegex);
+      const closeParentheses = part.match(this.closeParenthesesRegex);
+
+      // Ignore parentheses if they are opened and closed in the same condition as we put them using ifelse function
+      // Extra parentheses will throw error in Lens
+      // We keep the open parentheses and the closing parentheses that are not part of the same condition
+      if (
+        openParentheses &&
+        openParentheses.length > 0 &&
+        closeParentheses &&
+        closeParentheses.length > 0
+      ) {
+        return result;
+      }
+      // Put back the open parentheses
+      if (openParentheses && openParentheses.length > 0) {
+        result = openParentheses[0] + result;
+      }
+      // Put back the closing parentheses
+      if (closeParentheses && closeParentheses.length > 0) {
+        result = result + closeParentheses[0];
+      }
+      return result;
+    });
+
+    // At the end of the loop, we will have an array of conditions with AND(*) and OR(+)
+    // And to evaluate the whole condition, by wrapping them with ifelse and check if all the conditions are true (>0)
+    const conditions = res.join(' ') + ' > 0';
+    return conditions;
+  }
+
+  getTrueBranch(expression: string) {
+    const lastColonPosition = this.diveIntoString(expression, this.IF, this.ELSE);
+    if (lastColonPosition !== undefined) {
+      return expression.substring(expression.indexOf(this.IF) + 1, lastColonPosition);
+    }
+    return '';
+  }
+
+  getFalseBrach(expression: string) {
+    const lastColonPosition = this.diveIntoString(expression, this.IF, this.ELSE);
+    if (lastColonPosition !== undefined) {
+      return expression.substring(lastColonPosition + 1);
+    }
+    return '';
+  }
+
+  getCondition(expression: string): string {
+    return expression.substring(0, expression.indexOf(this.IF));
+  }
+
+  parse(): string {
+    const recursiveParseConditions = (expression: string): string => {
```

Comment:
To me, `condition` word looks closer to what it is parsing than `booleanLogic` when dealing with `if else` type of conditions.

Reply:
I see, in [conditional operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_operator), one part of the expression is called condition:
```
condition ? exprIfTrue : exprIfFalse
```
So, when I first checked this naming, I thought it would only parse the condition section, but it actually parses the whole expression.

`parseConditionalExpressions` seems a more clear name then, but I am fine with keeping the current one, too.